crates/goose-cli/src/commands/session.rs

```diff
@@ -44,9 +45,21 @@ pub fn build_session<'a>(
 
     let provider_config = get_provider_config(&loaded_profile.provider, (*loaded_profile).clone());
 
-    // TODO: Odd to be prepping the provider rather than having that done in the agent?
     let provider = factory::get_provider(provider_config).unwrap();
-    let agent = AgentFactory::create(agent_version.as_deref().unwrap_or("default"), provider).unwrap();
+    let mut agent =
+        AgentFactory::create(agent_version.as_deref().unwrap_or("default"), provider).unwrap();
+
+    // We now add systems to the session based on configuration
+    // TODO update the profile system tracking
+    // TODO use systems from the profile
```

Comment:
What's the plan here. Bring back some notion of `~/.config/goose/profiles.yaml`?

Reply:
they are already in there, but not used currently. we'll want to rework them to ensure the right content but should work